### PR TITLE
XRButton: allow specifying optional and required features

### PR DESCRIPTION
**Pull request by snowzurfer**
## Update `XRButton` to let user pass optional and required WebXR features

Update `XRButton` so that it can be passed optional and required features to be enabled on the `XRSession`. This is useful for enabling e.g. light estimation:
```javascript
const sessionInit = {
	optionalFeatures: [ 'light-estimation' ],
};
document.body.appendChild( XRButton.createButton( renderer, sessionInit ) );
```

### CODE REVIEW

**Comment by snowzurfer** [comment]
not sure if this is the right forum, but @snagy would you know if you folks at Meta have a timeline for implementing this in the Quest Browser?

**Comment by Mugen87** [comment]
Related example: https://threejs.org/examples/webxr_ar_lighting

Question: Do we need a new method like `getLightProbe()` or can `XREstimatedLight` from the addons be used for the same use cases?

**Comment by snowzurfer** [comment]
Neat, that looks like it already does what I wanted to do, and shows how to!

At this point that makes this PR kind of irrelevant.

I think I will repurpose this PR to support https://threejs.org/docs/#api/en/extras/PMREMGenerator
since I don't see it being used in the code.

**Comment by snowzurfer** [comment]
One thing we could still use this PR for is to support passing requested WebXR features through the XRButton.

**Comment by snowzurfer** [comment]
Updated the PR and the description.

**Review by Mugen87 — APPROVED** [review]

**Comment by Mugen87** [comment]
https://rawcdn.githack.com/snowzurfer/three.js/albtaiuti/webxr-light-probe-on-webxr-manager/examples/index.html

File: examples/jsm/webxr/XRButton.js
Comment (Mugen87) [inline examples/jsm/webxr/XRButton.js:3]: I can't merge this PR until @mrdoob approves since I'm not why `sessionInit` wasn't added in the first place. Looking at the code, the module request multiple optional features at once. Maybe this was done in order to keep the interface simple. Adding `sessionInit` means devs have to know how the `options` of `requestSession()` is structured.

**Comment by mrdoob** [comment]
I'll have to think about this one...

**Comment by Mugen87** [comment]
The merged PR https://github.com/mrdoob/three.js/pull/27041 relies on this one since it assume you can pass `light-estimation` to `XRButton`.

If we don't merge this PR or a similar solution, we should revert #27041. Otherwise it will confuse devs.

**Comment by mrdoob** [comment]
Oh true... Then I'll merge it.
All this is still very much WIP anyway...

**Pull request merged**
